v0/util/network.js

```diff
@@ -68,7 +68,7 @@ function handleResponseRules(responseRules, errorCode, url) {
 
 const getAxiosResponse = async (url, params, responseRules, errorMessage) => {
   const resp = await axios.get(url, params).catch(error => {
-    throw error;
+    throw new CustomError(error, 400);
```

Comment:
Can we implement something like this?

```javascript
axios.post(url, data).then(res => {
        // do good things
})
.catch(err => {
    if (err.response) {
      // client received an error response (5xx, 4xx)
	  // handle acordingly
    } else if (err.request) {
      // client never received a response, or request never left. we should retry, i.e. return 5xx
    } else {
      // anything else
    }
})
```

Ref: https://www.intricatecloud.io/2020/03/how-to-handle-api-errors-in-your-web-app-using-axios/